lms/djangoapps/grades/signals/handlers.py

```diff
@@ -144,6 +145,8 @@ def score_published_handler(sender, block, user, raw_earned, raw_possible, only_
             modified=score_modified_time,
             score_db_table=ScoreDatabaseTableEnum.courseware_student_module,
         )
+        if isinstance(block, ScorableXBlockMixin):
+            block.set_score(Score(raw_earned=raw_earned, raw_possible=raw_possible))
```

Comment:
Indeed.  `set_score()` probably shouldn't be responsible for that (because then every block has to duplicate the effort).  It 's currently happening in the instructor task helper (immediately after calling `rescore()`), but since we've moved the logic over to the grades infrastructure, I think it makes sense for the persistence to happen there too.